lib/workers/evm_server.rb

```diff
@@ -77,6 +78,18 @@ def set_process_title
     Process.setproctitle(SERVER_PROCESS_TITLE) if Process.respond_to?(:setproctitle)
   end
 
+  def set_database_application_name
+    name = database_application_name
+
+    # disconnect! after querying the information needed to set the application name
+    ActiveRecord::Base.connection_pool.disconnect!
```

Comment:
This happens only once in the server process.  I'm not quite sure why `SET application_name ...` doesn't work here on the existing connection.